box-browse-sdk/src/main/java/com/box/androidsdk/browse/adapters/BoxItemAdapter.java

```diff
@@ -111,7 +114,41 @@ public void run() {
         return index;
     }
 
+    /**
+     * Does the appropriate add and removes to display only provided items.
+     * @param items
+     */
+    public synchronized void updateTo(ArrayList<BoxItem> items){
 
+        final HashMap<String, Integer> oldPositionMap = mItemsPositionMap;
+        final Set<String> oldIds = oldPositionMap.keySet();
+        mItemsPositionMap = new HashMap<String, Integer>(items.size());
+        mItems.clear();
+        addAll(items);
+        if (oldPositionMap.size() > mItems.size() ){
+
+            Iterator<Map.Entry<String,Integer>> iterator = oldPositionMap.entrySet().iterator();
+            while (iterator.hasNext()){
+                Map.Entry<String,Integer> entry = iterator.next();
+                if (mItemsPositionMap.containsKey(entry.getKey())){
+                    iterator.remove();
+                }
+            }
+            mHandler.post(new Runnable() {
+                @Override
+                public void run() {
+                    for (Map.Entry<String, Integer> entry : oldPositionMap.entrySet()){
+                        notifyItemRemoved(entry.getValue());
```

Comment:
Does this actually show the items removed individually? Otherwise it seems like wasted effort since we call notifyItemRange right after